runsc/cmd/tar.go

```diff
@@ -122,7 +118,7 @@ func (r *RootfsUpper) Execute(ctx context.Context, f *flag.FlagSet, args ...any)
 		defer out.Close()
 	}
 
-	if err := c.Sandbox.TarRootfsUpperLayer(out); err != nil {
+	if err := c.Sandbox.TarRootfsUpperLayer(id, out); err != nil {
```

Comment:
I think we should also define `TarRootfsUpperLayer()` method on `Container`, which would then call `c.Sandbox.TarRootfsUpperLayer(c.ID, out)`. That would be cleaner. Then in the tests you can simply do `containers[0].TarRootfsUpperLayer(rootTar)` instead of `containers[0].Sandbox.TarRootfsUpperLayer(containers[0].ID, rootTar)`